src/lv_core/lv_indev.c

```diff
@@ -1053,42 +1055,20 @@ static void indev_drag(lv_indev_proc_t * state)
             /*Get the coordinates of the object and modify them*/
             lv_coord_t act_x      = lv_obj_get_x(drag_obj);
             lv_coord_t act_y      = lv_obj_get_y(drag_obj);
-            uint16_t inv_buf_size = lv_disp_get_inv_buf_size(
-                indev_act->driver.disp); /*Get the number of currently invalidated areas*/
-
-            lv_coord_t prev_x     = drag_obj->coords.x1;
-            lv_coord_t prev_y     = drag_obj->coords.y1;
-            lv_coord_t prev_par_w = lv_obj_get_width(lv_obj_get_parent(drag_obj));
-            lv_coord_t prev_par_h = lv_obj_get_height(lv_obj_get_parent(drag_obj));
 
             if(allowed_dirs & LV_DRAG_DIR_HOR)
-            	lv_obj_set_x(drag_obj, act_x + state->types.pointer.vect.x);
+                lv_obj_set_x(drag_obj, act_x + state->types.pointer.vect.x);
 
             if(allowed_dirs & LV_DRAG_DIR_VER)
-            	lv_obj_set_y(drag_obj, act_y + state->types.pointer.vect.y);
-
-            /*Set the drag in progress flag if the object is really moved*/
-            if(drag_obj->coords.x1 != prev_x || drag_obj->coords.y1 != prev_y) {
-                if(state->types.pointer.drag_in_prog !=
-                   0) { /*Send the drag begin signal on first move*/
-                    drag_obj->signal_cb(drag_obj, LV_SIGNAL_DRAG_BEGIN, indev_act);
-                    if(state->reset_query != 0) return;
-                }
-                state->types.pointer.drag_in_prog = 1;
-            }
-            /*If the object didn't moved then clear the invalidated areas*/
-            else {
-                /*In a special case if the object is moved on a page and
-                 * the scrollable has fit == true and the object is dragged of the page then
-                 * while its coordinate is not changing only the parent's size is reduced */
-                lv_coord_t act_par_w = lv_obj_get_width(lv_obj_get_parent(drag_obj));
-                lv_coord_t act_par_h = lv_obj_get_height(lv_obj_get_parent(drag_obj));
-                if(act_par_w == prev_par_w && act_par_h == prev_par_h) {
-                    uint16_t new_inv_buf_size = lv_disp_get_inv_buf_size(indev_act->driver.disp);
-                    lv_disp_pop_from_inv_buf(indev_act->driver.disp,
-                                             new_inv_buf_size - inv_buf_size);
-                }
+                lv_obj_set_y(drag_obj, act_y + state->types.pointer.vect.y);
+
+            /*Set the drag in progress flag*/
+            /*Send the drag begin signal on first move*/
+            if(state->types.pointer.drag_in_prog == 0) {
```

Comment:
@kisvegabor This condition was previously inverted:

https://github.com/littlevgl/lvgl/blob/c58fd97cf7f35b5091342f116a3c2b317eb434e0/src/lv_core/lv_indev.c#L1067-L1068

This seemed incorrect to me. It would cause `LV_SIGNAL_DRAG_BEGIN` to be sent repeatedly while dragging.

Reply:
Why? It is sent only when `drag_in_prog` (drag in progress) is zero. When drag is in progress the flag is set to 1.

Reply:
The original file, which I linked to in my comment (`Lines 1067 to 1068 in c58fd97`), sends `LV_SIGNAL_DRAG_BEGIN` when drag_in_prog is *not equal* to 0.

It's different than the patch that GitHub is showing with the green background.

Reply:
Oh really, my bad.